src/test/java/automationpractice/com/pages/ProductDetailsPage.java

```diff
@@ -0,0 +1,215 @@
+package automationpractice.com.pages;
+
+import automationpractice.com.exceptions.NoColorAvailableException;
+import automationpractice.com.exceptions.NoSizeAvailableException;
+import net.serenitybdd.core.pages.PageObject;
+import net.serenitybdd.core.pages.WebElementFacade;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ProductDetailsPage extends PageObject {
+
+    private final String SELECT_SIZE_XPATH_PATTERN = "//*[@id='group_1']//option[@title='%s']";
+    private final String SELECT_COLOR_XPATH_PATTERN = "//*[@id='color_to_pick_list']//a[@title='%s']";
```

Comment:
static

Reply:
1. Czy na pewno musimy dodawać tu static?  Z tych zmiennych korzystamy tylko w tej klasie, nigdzie na zewnątrz tego nie potrzebujemy.
2. Po zmianie na static zacznie rzucać błędem z GoogleStyle.xml "'static' modifier out of order with the JLS suggestions."